File: pset6/mario/mario.py

height = input("Height: ")

# Prompt for height, height must be an integer and between or equal to 1 and 8
while (not height.isdigit() or int(height) > 8 or int(height) < 1):   
    height = input("Height: ")

# convert str to int
height = int(height)


# print pattern
for i in range(1, height+1):
    for j in range(height - i):
        print(" ", end="")
    
    for k in range(i):
        print("#", end="")
    
    print("  ", end="")
    
    for k in range(i):
        print("#", end="")
        
    print("")

File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Prompt for height, height must between or equal to 1 and 8
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height > 8 || height < 1);
    
    // Print required pattern
    int i, j, k;

    // Print row i
    for (i = 1; i <= height; i++)
    {
        // Print spaces before
        for (j = height - i; j > 0; j--)
        {
            printf(" ");
        }
        
        // Print left blocks 
        for (k = 0; k < i; k++)
        {
            printf("#"); 
        }
        
        // Print middle spaces
        printf("  ");
        
        // Print right blocks
        for (k = 0; k < i; k++)
        {
            printf("#"); 
        }
        
        // Change line
        printf("");
    }
}